apps/agentstack-server/src/agentstack_server/infrastructure/text_extraction/docling.py

```diff
@@ -44,19 +89,37 @@ async def extract_text(self, *, file_url: AnyUrl, timeout: timedelta | None = No
         ):
             response.raise_for_status()
 
-            md_stream = None
-
-            async def read(chunk_size: int = 1024) -> bytes:
-                nonlocal md_stream
-                if not md_stream:
-                    md_stream = extract_string_value_stream(response.aiter_text, "md_content", chunk_size)
-                async for text_chunk in md_stream:
-                    return text_chunk.encode("utf-8")
-                return b""
-
-            yield AsyncFile(
-                filename="extracted_text.md",
-                content_type="text/markdown",
-                read=read,
-                size=None,  # size is unknown beforehand
-            )
+            # Build a dict of keys to extract with their types
+            keys_to_extract = {info.field_key: "object" if info.is_object else "string" for _, info in format_pairs}
+
+            # Extract all requested formats in a single pass through the stream
+            extracted_contents = await extract_multiple_values_stream(response.aiter_text, keys_to_extract)
+
+            # Create (AsyncFile, ExtractionFormat) tuples for each format
+            files = []
+            for extraction_format, info in format_pairs:
+                content = extracted_contents[info.field_key]
+
+                # Create a read function that returns the buffered content
+                def create_read_fn(buffered_content: bytes):
```

Comment:
This is the full file in bytes right? That defies the purpose of streaming altogether, the point of this and the streaming parser was to avoid loading the file to memory :)

Notice the previous function `extract_string_value_stream` was "stream in" -> "stream out".

`extract_multiple_values_stream` is "stream in" -> "loaded extracted content out" or am I missing something?